diff -urN pygtail-0.8.0.org/pygtail/core.py pygtail-0.8.0/pygtail/core.py
--- pygtail-0.8.0.org/pygtail/core.py	2018-03-10 17:34:16.000000000 +0100
+++ pygtail-0.8.0/pygtail/core.py	2018-04-05 09:26:52.894624325 +0200
@@ -182,7 +182,7 @@
         if self.on_update:
             self.on_update()
         offset = self._filehandle().tell()
-        inode = stat(self.filename).st_ino
+        inode = fstat(self._filehandle().fileno()).st_ino
         fh = open(self._offset_file, "w")
         fh.write("%s\n%s\n" % (inode, offset))
         fh.close()
